compiler/rustc_traits/src/type_op.rs

```diff
@@ -59,12 +59,14 @@ pub fn type_op_ascribe_user_type_with_span<'tcx>(
 
     let UserSubsts { user_self_ty, substs } = user_substs;
     let tcx = ocx.infcx.tcx;
+    let misc_cause = ObligationCause::misc(span, hir::CRATE_HIR_ID);
+    let dummy_cause = ObligationCause::dummy_with_span(span);
```

Comment:
What happens if you just use `misc_cause` everywhere?

Reply:
Ohh yeah, it's equivalent from what I see in code.